src/components/UserProfile/UserProfileAccount/index.js

```diff
@@ -1,53 +1,78 @@
-import React, { useState } from "react";
+import React, { useEffect, useState } from "react";
+import { connect } from "react-redux";
 import Grid from "@material-ui/core/Grid";
 import { withStyles } from "@material-ui/styles";
-import { connect } from "react-redux";
-import isEmpty from "lodash/isEmpty";
+import map from "lodash/map";
+import find from "lodash/find";
+import lowerCase from "lodash/lowerCase";
 
 import StyledDropdown from "../../common/StyledDropdown";
 import { useStyles } from "./styles";
-import { walletTypes } from "../../../Redux/actionCreators/UserActions";
-import { userActions } from "../../../Redux/actionCreators";
+import {
+  fetchAvailableUserWallets,
+  fetchWalletLinkedProviders,
+  walletTypes,
+} from "../../../Redux/actionCreators/UserActions";
 import MetamaskDetails from "./MetamaskDetails";
 import { initSdk } from "../../../utility/sdk";
+import ProviderBalance from "./ProviderBalance";
 import AlertBox, { alertTypes } from "../../common/AlertBox";
+import ProvidersLinkedCount from "./ProvidersLinkedCount";
+import { startAppLoader, stopAppLoader } from "../../../Redux/actionCreators/LoaderActions";
+import { LoaderContent } from "../../../utility/constants/LoaderContent";
 
-const walletDropdownList = Object.entries(walletTypes).map(([key, value]) => ({ value, label: key }));
-
-const UserProfileAccount = ({ classes, wallet, updateWallet, stopWalletDetailsPolling }) => {
+const UserProfileAccount = ({ classes, dispatch }) => {
   const [alert, setAlert] = useState({});
+  const initialWallet = { value: "default", type: "default" };
+  const [wallet, updateWallet] = useState(initialWallet);
+  const [wallets, updateWallets] = useState([]);
+  const [linkedProviders, updateLinkedProviders] = useState([]);
+  useEffect(() => {
+    const fetchWallets = async () => {
+      const availableWallets = await fetchAvailableUserWallets();
```

Comment:
we have to dispatch actions ryt for making API calls..

Reply:
Let's track this and address it separately.

Reply:
https://github.com/singnet/snet-dapp/pull/415